README.md

```diff
@@ -95,10 +97,35 @@ After your Docker container successfully builds and is running locally you can s
-
-Currently the setup deploys both a data ingestion cloud funtion and a data ingestion cloud run instance. These are duplicates of each other. Once we get the cloud run one fully working we will delete the cloud fuction, but for now if you want to not have a duplicate while developing you can just comment out the setup for whichever one you don't want to use in `prototype.tf`
-
-Note that currently terraform doesn't diff the contents of the functions/cloud run service, so to get them to redeploy you can either
+Before deploying, make sure you have installed Terraform and a Docker client (e.g. Docker Desktop). See [One time setup](#one-time-setup) above.
+1. Create your own `terraform.tfvars` file in the same directory as the other terraform files. For each variable declared in `prototype_variables.tf` that doesn't have a default, add your own for testing. Typically your own variables should be unique and can just be prefixed with your name or ldap. There are some that have specific requirements like project ids, code paths, and image paths.
+2. Configure docker to use credentials through gcloud.  
+```gcloud auth configure-docker```
+3. On the command line, navigate to your project directory.  
+```cd path/to/your/project```
+4. Build and push your Docker images to Google Container Registry. Do this for both the run_ingestion and run_gcs_to_bq services.  
+   ```bash
+   # Build the image locally
+   docker build -t gcr.io/<project-id>/<your-image-name> run_ingestion/
+   
+   # Upload the image to Google Container Registry
+   docker push gcr.io/<project-id>/<your-image-name>
+   ```
+5. Deploy via Terraform.  
+   ```bash
+   # Get the latest image digests
+   export TF_VAR_run_ingestion_image_path=$(gcloud container images describe gcr.io/<project-id>/<your-ingestion-image> \
+   --format="value(image_summary.fully_qualified_digest)")
+   export TF_VAR_run_gcs_to_bq_image_path=$(gcloud container images describe gcr.io/<project-id>/<your-gcs-to-bq-image> \
+   --format="value(image_summary.fully_qualified_digest)")
+   
+   # Deploy via terraform, providing the paths to the latest images so it knows to redeploy
+   terraform apply -var="run_ingestion_image_path=$TF_VAR_run_ingestion_image_path" \
+   -var="run_gcs_to_bq_image_path=$TF_VAR_run_gcs_to_bq_image_path
+   ```
+
+Note: Currently the setup deploys both a cloud funtion and a cloud run instance for each pipeline. These are duplicates of each other. Eventually, we will delete the cloud fuctions, but for now you can just comment out the setup for whichever one you don't want to use in `prototype.tf`
+
+Note: Terraform doesn't diff the contents of the functions/cloud run service, so to redeploy them you can either
```

Comment:
Is it still true that you need to call taint or destroy for the cloud run services, or does the above setup take care of that?

Reply:
Added a third option for cloud run services.